### ISSUE TITLE
upload() doesn't convert file to an array with input multiple

### ISSUE THREAD

**Issue by tozaicevas** [testing-library/user-event#529]
Could someone explain to me the logic behind this line of code: https://github.com/testing-library/user-event/blob/master/src/upload.js#L17 ?

Consider this code snippet:

```
  function handleChooseImage(event: ChangeEvent<HTMLInputElement>) {
    const selectedFiles = event.target.files

    if (!selectedFiles?.length) return
   
    ...
  }

      <input
        data-testid="add-photos-input"
        ref={inputRef}
        type="file"
        onChange={handleChooseImage}
        accept="image/*"
        multiple
        hidden
      />
```

My goal is to write following lines of test code
```
      const imageFile = new File([], 'imgname', { type: 'image/png' })
      const imageInput = screen.getByTestId('add-photos-input')

      userEvent.upload(imageInput, imageFile)
```

This test code result in event handler being called with an object, where `length` is `undefined` and `item()` returns `undefined`:

`selectedFiles:  { length: undefined, item: [Function: item] }` 

Is that expected behaviour?

**Comment by ph-fritsche** [comment]
The function expects you to provide `File` for single file inputs and `File[]` for multiple file inputs.
```js
userEvent.upload(multipleFileInput, [imageFile])
```
I think this could be improved by just converting it.

**Comment by ph-fritsche** [comment]
resolved in [12.6.2](https://github.com/testing-library/user-event/releases/tag/v12.6.2)
